tools/cli/pkg/command/completion.go

```diff
@@ -0,0 +1,119 @@
+package command
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+// CompletionCommand represents an execution of the kcp completion command
+type CompletionCommand struct {
+	cobraCmd       *cobra.Command
+	completeTarget string
+}
+
+var validArguements = []string{"bash", "zsh", "fish", "powershell"}
+
+const errorMsg = "accepts 1 arg(s), received 0 \n"
+const suggestionMsg = "Please use `kcp completion [bash|zsh|fish|powershell]`"
+const savedInMsg = "Saved in"
+const savedFileName = "kcp_completion"
+
+// NewCompletionCmd constructs a new instance of CompletionCommand and configures it in terms of a cobra.Command
+func NewCompletionCommand() *cobra.Command {
+	cmd := CompletionCommand{}
+	cobraCmd := &cobra.Command{
+		Use:   "completion [bash|zsh|fish|powershell]",
+		Short: "Generate completion script",
+		Long: `To load completions:
+
+Bash:
+
+$ source <(kcp completion bash)
+
+# To load completions for each session, execute once:
+Linux:
+  $ kcp completion bash > /etc/bash_completion.d/kcp
+MacOS:
+  $ kcp completion bash > /usr/local/etc/bash_completion.d/kcp
+
+Zsh:
+
+# If shell completion is not already enabled in your environment you will need
+# to enable it.  You can execute the following once:
+
+$ echo "autoload -U compinit; compinit" >> ~/.zshrc
+
+# To load completions for each session, execute once:
+$ kcp completion zsh > "${fpath[1]}/_kcp"
+
+# You will need to start a new shell for this setup to take effect.
+
+Fish:
+
+$ kcp completion fish | source
+
+# To load completions for each session, execute once:
+$ kcp completion fish > ~/.config/fish/completions/kcp.fish
+
+Powershell:
+
+PS> kcp completion powershell | Out-String | Invoke-Expression
+
+# To load completions for every new session, run:
+PS> kcp completion powershell > kcp.ps1
+# and source this file from your powershell profile.
```

Comment:
This section didn't appear correctly in the .md file:
<img width="300" alt="Screen Shot 2021-01-27 at 18 48 19" src="https://user-images.githubusercontent.com/49867570/106032128-60e4bf00-60d0-11eb-8f64-26ce9676fc68.png">
 because the `#` symbol denotes a Title in Markdown. I tried to reorganize the text. 
``````suggestion
		Long: `Generates completion script for Bash, Zsh, Fish, or Powershell.

### Bash

To load completions for Bash, run:
```bash
$ source <(kcp completion bash)
```
To load completions for each session, execute once:

- Linux:
  ```bash
    $ kcp completion bash > /etc/bash_completion.d/kcp
  ```

- MacOS:
  ```bash
    $ kcp completion bash > /usr/local/etc/bash_completion.d/kcp
  ```

### Zsh

If shell completion is not already enabled in your environment you will need to enable it.  You can execute the following once:
```zsh
$ echo "autoload -U compinit; compinit" >> ~/.zshrc
```

To load completions for each session, execute once:
```zsh
$ kcp completion zsh > "${fpath[1]}/_kcp"
```

You will need to start a new shell for this setup to take effect.

### Fish

To load completions for Fish, run:
```shell
$ kcp completion fish | source
```

To load completions for each session, execute once:
```shell
$ kcp completion fish > ~/.config/fish/completions/kcp.fish
```

### Powershell

To load completions for Powershell, run:
```shell
PS> kcp completion powershell | Out-String | Invoke-Expression
```

To load completions for every new session, run:
```shell
PS> kcp completion powershell > kcp.ps1
```
Source this file from your Powershell profile.
``````

Reply:
followed and updated